src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -527,6 +539,18 @@ public override bool UnmapEnum<TEnum>(string? pgName = null, INpgsqlNameTranslat
         return found;
     }
 
+    public override bool UnmapEnum(Type clrType, string? pgName = null, INpgsqlNameTranslator? nameTranslator = null)
```

Comment:
Take a look at how UnmapComposite is implemented - the generic version actually calls the non-generic version, let's do it the same way with enums.

Reply:
The thing is, in the case of UnmapComposite() for ConnectorTypeMapper the non-generic version of the method was created earlier than the generic one. And for UnmapEnum() the generic version was first - so the current (working) implementation relies on those generics. To be specific - it relies on UserEnumTypeMapping<TEnum> class.

I see the solution either by using Activator or by implementing a non-generic version of the UserEnumTypeMapping class. But either way, if we do this, we will use reflection in the generic versions of UnmapEnum(), which is not optimal, since the generic version was introduced to be as compile-time safe as possible.

In my current implementation, we use reflection only in case the user decided to take this route - by invoking a non-generic method.

Reply:
We can't use the same approach for this method (at least without a rewrite). It relies on generic 
UserEnumTypeMapping<TEnum> class: 

`
    public override bool UnmapEnum<TEnum>(string? pgName, INpgsqlNameTranslator? nameTranslator)
    {
        .....
        var userEnumMapping = new UserEnumTypeMapping<TEnum>(pgName, nameTranslator);
        .....
    }`

And this class relies on the generic EnumHandler<TEnum>. Rewriting those to support non-generic enum mapping seems like a big task.

An alternative would be to instantiate this generic class with Activator. This way the generic version would call the non-generic version, but then we will have run-time reflection even when users call the generic version of the method, which is not necessary.